Source/Fuse.Controls.CameraView/iOS/CameraViewImpl.mm

```diff
@@ -400,6 +401,116 @@ void setCameraFacing(CameraView* cameraView, int cameraFacing, void(^onResolve)(
 		});
 	}
 
+
+	void setCameraFocusPoint(CameraView* cameraView, 
+		double x, double y, int cameraWidth, int cameraHeight, 
+		int isFocusLocked,
+		void(^onResolve)(int), void(^onReject)(NSString*)) {
+
+		AVState* avState = cameraView->avState;
+		dispatch_async(avState->queue, ^{
+
+			AVCaptureDevice* currentDevice = avState->deviceInput.device;
+
+			if(!currentDevice)
+				return;
+			
+			if([currentDevice isFocusPointOfInterestSupported] 
+				&& [currentDevice isFocusModeSupported:AVCaptureFocusModeContinuousAutoFocus]
+				&& [currentDevice isExposureModeSupported:AVCaptureExposureModeContinuousAutoExposure]
+				)
+			{
+				NSError* error;
+
+				//check to unlock
+				if (isFocusLocked != 1 && [currentDevice lockForConfiguration:&error])
+				{
+
+					[currentDevice setFocusMode:AVCaptureFocusModeContinuousAutoFocus];
+			    [currentDevice setExposureMode:AVCaptureExposureModeContinuousAutoExposure];
+					[currentDevice setWhiteBalanceMode:AVCaptureWhiteBalanceModeContinuousAutoWhiteBalance];
+			    [currentDevice unlockForConfiguration];
+
+					[avState->session commitConfiguration];
+				}
+
+
+				if([currentDevice lockForConfiguration:&error])
+				{
+					CGPoint focusPoint;
+
+			    float focus_x = x/cameraWidth;
+			    float focus_y = y/cameraHeight;
+
+
+					switch([[UIDevice currentDevice] orientation]) {
+						case UIDeviceOrientationPortrait:
+							focusPoint = CGPointMake(focus_y, 1.f - focus_x);
+							break;
+						case UIDeviceOrientationPortraitUpsideDown:
+							focusPoint = CGPointMake(1.f - focus_y, focus_x);
+							break;
+						case UIDeviceOrientationLandscapeLeft:
+							focusPoint = CGPointMake(focus_y, 1.f - focus_x);
+							break;
+						case UIDeviceOrientationLandscapeRight:
+							focusPoint = CGPointMake(1.f - focus_x, focus_y);
+							break;
+						default:
+							focusPoint = CGPointMake(focus_y, 1.f - focus_x);
+							break;
+					}
+
+
+			    if ([currentDevice isFocusPointOfInterestSupported]) {
+			    	[currentDevice setFocusPointOfInterest:focusPoint];	
+			    }
+					
+					if ([currentDevice isExposurePointOfInterestSupported]) {
+				    [currentDevice setExposurePointOfInterest:focusPoint];
+				  }
+
+					[currentDevice setFocusMode:AVCaptureFocusModeContinuousAutoFocus];
+			    [currentDevice setExposureMode:AVCaptureExposureModeContinuousAutoExposure];
```

Comment:
Seems something went wrong with the indentation here... In general, this commit seems to introduce a lot of whitespace issues in this file... It seems this is due to inconsistent indentation style. Please fix that.

Just a tip: I would suggest coding with visible white-space so you can see what's going on...

Reply:
Yeah I dunno how that happened, looks fine on my side, will go through each line...